app/src/main/java/com/hover/stax/data/repository/BountyRepositoryImpl.kt

```diff
@@ -0,0 +1,89 @@
+package com.hover.stax.data.repository
+
+import com.hover.sdk.actions.HoverAction
+import com.hover.sdk.sims.SimInfo
+import com.hover.stax.channels.Channel
+import com.hover.stax.countries.CountryAdapter
+import com.hover.stax.data.local.actions.ActionRepo
+import com.hover.stax.domain.model.Bounty
+import com.hover.stax.domain.model.ChannelBounties
+import com.hover.stax.domain.repository.BountyRepository
+import com.hover.stax.transactions.StaxTransaction
+import com.uxcam.internals.an.t
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+class BountyRepositoryImpl(val actionRepo: ActionRepo, private val coroutineDispatcher: CoroutineDispatcher) : BountyRepository {
+
+    override val bountyActions: List<HoverAction>
+        get() = actionRepo.bounties
+
+    override fun simPresent(bounty: Bounty, sims: List<SimInfo>): Boolean {
+        if (sims.isEmpty()) return false
+
+        sims.forEach { simInfo ->
+            for (i in 0 until bounty.action.hni_list.length()) if (bounty.action.hni_list.optString(i) == simInfo.osReportedHni) return true
+        }
+
+        return false
+    }
+
+    override fun getCountryList(): Flow<List<String>> = channelFlow {
+        launch(coroutineDispatcher) {
+            val actions = bountyActions
+
+            val countryCodes = mutableListOf(CountryAdapter.CODE_ALL_COUNTRIES)
+            actions.asSequence().map { it.country_alpha2.uppercase() }.distinct().sorted().toCollection(countryCodes)
+
+            send(countryCodes)
+        }
+    }
+
+    override suspend fun makeBounties(actions: List<HoverAction>, transactions: List<StaxTransaction>?, channels: List<Channel>): List<ChannelBounties> {
```

Comment:
It's okay as it is, Do you think we should consider having a  Factory? BountyFactory for example; so that may handle the multiple ways an object can be made.

Reply:
Factories are for when you aren't sure what class of object needs to be made. This always makes a bounty, it is just filtering which transactions are actually part of it.